app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt

```diff
@@ -509,18 +508,22 @@ class DefaultSessionControlController(
         HomeScreen.customizeHomeClicked.record(NoExtras())
     }
 
-    override fun handleShowWallpapersOnboardingDialog(state: WallpaperState): Boolean {
-        return state.availableWallpapers.filter { wallpaper ->
-            wallpaper.thumbnailFileState == Wallpaper.ImageFileState.Downloaded
-        }.size.let { downloadedCount ->
-            // We only display the dialog if enough thumbnails have been downloaded for it.
-            downloadedCount >= THUMBNAILS_SELECTION_COUNT
-        }.also { showOnboarding ->
-            if (showOnboarding) {
-                navController.nav(
-                    R.id.homeFragment,
-                    HomeFragmentDirections.actionGlobalWallpaperOnboardingDialog(),
-                )
+    override fun handleShowWallpapersOnboardingDialog(state: AppState): Boolean {
+        return if (state.mode == Mode.Private) {
```

Comment:
```suggestion
        return if (activity.browsingModeManager.mode.isPrivate) {
```
We don't need to pass the `AppState` here since `SessionControlController` has the `AppStore` as a param so let's avoid that. 

However, based on my search on this file, we seem to also check if it is private mode via the code above in this file. We use this `isPrivate` in quick a few places more than the current `appState.mode` while I don't believe there should be a difference. I would be more comfortable with the `isPrivate` approach given the existing general usage for the uplift. I would assume `appState.mode` should just work as well, but this is making me think whether or not we even need `browsingModeManager` anymore, and whether or not we can refactor somethings. So, I would prefer that investigation to be a followup.